snapcraft/projects.py

```diff
@@ -39,7 +36,7 @@ class Config:  # pylint: disable=too-few-public-methods
 
         validate_assignment = True
         extra = "allow"  # FIXME: change to 'forbid' after model complete
-        allow_mutation = False
+        allow_mutation = True
```

Comment:
We update project fields with adopted metadata to take advantage of pydantic's field assignment validation.

Reply:
I don't understand this comment, maybe a more clear code comment would be better